sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java

```diff
@@ -333,6 +333,8 @@ private OpenConnectionRntbdRequestRecord processWhenConnectionOpened(
             // Releasing the channel back to the pool so other requests can use it
             this.releaseToPool(channel);
 
+            requestRecord.getAddressUri().setConnected();
```

Comment:
I see that we don't update the replica health when it is not success. Is that because the request can fail at transport layer because of any reason? And that doesn't necessarily mean that it is a replica issue?

Reply:
yes, currently we only mark the replica as unhealthy when it triggers an address refresh. 
And also if the request failed at connection, then internally SDK will retry the request with forceRefresh to be true.